statie/source/_posts/2019/2019-05-21-review-of-the-refactoring.md

```diff
@@ -37,10 +37,62 @@ Almost all refactorings include inverse refactoring.
 When there is [Extract Variable](https://refactoring.com/catalog/extractVariable.html), you can find also [Inline Variable](https://refactoring.com/catalog/inlineVariable.html).
 That was a surprise for me as I thought the first option is always better (and I learned that's not true).
 
+Extracting variable gives a name to the statement
+```php
+return preg_replace('/(?:[,.]0+|([,.]\d*?)0+)$/', '$1', $value);
+
+// refactored to
+$croppingZerosPattern = '/(?:[,.]0+|([,.]\d*?)0+)$/';
+return preg_replace($croppingZerosPattern, '$1', $value);
```

Comment:
Nice!
Why not constant?

Reply:
Because the refactoring is called "extract **variable**". Could be constant and doesn't matter much for article purposes :-)

Reply:
Tha's very bad reason to do anything :D 
Ok, not problem

Reply:
You are right, the reason is bad... then the example is bad.... But I'll keep it :-D